ui/plugins/eu.esdihumboldt.hale.ui/src/eu/esdihumboldt/hale/ui/io/IOWizard.java

```diff
@@ -517,133 +521,148 @@ public boolean performFinish() {
 		if (!applyConfiguration()) {
 			return false;
 		}
+		FileIOSupplier source = (FileIOSupplier) ((ImportProvider) getProvider()).getSource();
```

Comment:
Is that really a safe cast?
What about non-file locations like HTTP/HTTPS or JDBC URIs?

Reply:
Code updated for the non-file locations.